dartagnan/src/main/java/com/dat3m/dartagnan/verification/solving/RefinementSolver.java

```diff
@@ -124,12 +121,13 @@ private Result run() throws InterruptedException, SolverException, InvalidConfig
         performStaticWmmAnalyses(task, analysisContext, config);
         performStaticWmmAnalyses(baselineTask, baselineContext, config);
 
-        ProgramEncoder programEncoder = ProgramEncoder.fromConfig(program, analysisContext, config);
-        PropertyEncoder propertyEncoder = PropertyEncoder.fromConfig(program, baselineModel, analysisContext, config);
+        context = EncodingContext.of(task, analysisContext, ctx);
```

Comment:
Constructing the encoding context with the baseline task should work, no?

Reply:
It seems to work.